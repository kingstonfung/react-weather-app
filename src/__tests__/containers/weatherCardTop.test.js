import React from 'react';
import ReactDOM from 'react-dom';

import WeatherCardTop from '../../containers/WeatherCardTop';

describe('WeatherCard component...', () => {
  const staticWeatherCardBottom =
    <WeatherCardTop
      temperature="10"
      currentCondition="rain"
      city="Somewhere"
      cityImage="example.com/testimage.jpg"
    />;

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should create the background image inline style correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div.firstChild.getAttribute('style')).toBe('background-image: url(example.com/testimage.jpg);');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});