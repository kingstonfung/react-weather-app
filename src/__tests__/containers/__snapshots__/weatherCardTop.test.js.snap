// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component... should match snapshot 1`] = `
<div>
  <div
    class="cardUpperPortion"
    style="background-image: url(example.com/testimage.jpg);"
  >
    <div
      class="cardInfo"
    >
      <div
        class="cardTemperature"
      >
        10
      </div>
      <div
        class="currentCondition"
      >
        rain
      </div>
    </div>
    <h2
      class="cityNameTitle"
    >
      Somewhere
    </h2>
  </div>
</div>
`;
